src/main/java/net/sf/jabref/model/database/DuplicationChecker.java

```diff
@@ -16,33 +17,34 @@
     // use a map instead of a set since i need to know how many of each key is in there
     private final Map<String, Integer> allKeys = new HashMap<>();
 
-    //##########################################
-    //  usage:
-    //  isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
-    //############################################
-    // if the newkey already exists and is not the same as oldkey it will give a warning
-    // else it will add the newkey to the to set and remove the oldkey
-    public boolean checkForDuplicateKeyAndAdd(String oldKey, String newKey) {
-        // LOGGER.debug(" checkForDuplicateKeyAndAdd [oldKey = " + oldKey + "] [newKey = " + newKey + "]");
+    /**
+     * Usage:
+     * <br>
+     * isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
+     *
+     * If the newkey already exists and is not the same as oldkey it will give a warning
+     * else it will add the newkey to the to set and remove the oldkey
+     *
+     * @return true, if there is a duplicate key, else false
+     */
+    public boolean checkForDuplicateKeyAndAdd(Optional<String> oldKey, Optional<String> newKey) {
```

Comment:
Related discussion:
http://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments



2016-08-26 8:25 GMT+02:00 Stefan Kolb <notifications@github.com>:

> In src/main/java/net/sf/jabref/model/database/DuplicationChecker.java
> <https://github.com/JabRef/jabref/pull/1577#discussion_r76369435>:
>
> > -    //############################################
> > -    // if the newkey already exists and is not the same as oldkey it will give a warning
> > -    // else it will add the newkey to the to set and remove the oldkey
> > -    public boolean checkForDuplicateKeyAndAdd(String oldKey, String newKey) {
> > -        // LOGGER.debug(" checkForDuplicateKeyAndAdd [oldKey = " + oldKey + "] [newKey = " + newKey + "]");
> > +    /**
> > +     * Usage:
> > +     * <br>
> > +     * isDuplicate=checkForDuplicateKeyAndAdd( null, b.getKey() , issueDuplicateWarning);
> > +     *
> > +     * If the newkey already exists and is not the same as oldkey it will give a warning
> > +     * else it will add the newkey to the to set and remove the oldkey
> > +     *
> > +     * @return true, if there is a duplicate key, else false
> > +     */
> > +    public boolean checkForDuplicateKeyAndAdd(Optional<String> oldKey, Optional<String> newKey) {
>
> Not the exacr one but what about these: https://dzone.com/articles/
> use-of-optional-is-optional, https://dzone.com/articles/
> java-8-optional-how-use-it
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/JabRef/jabref/pull/1577/files/f9d6a1943a7bc023d8042f02c3c3922c38f34a66#r76369435>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATi5Fj7w1jmaRe-PdgqNDMDsD1MeDS2ks5qjobDgaJpZM4JKeSe>
> .
>